### PR TITLE
upload smoke test script for go client

### PR DESCRIPTION
**Pull request by lyiyu66**

### CODE REVIEW

File: autotest/client/go/goclienttest/testdata/Config/configDefaultLocaleEmpty.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+initializeCache: true
+#unit: second
+cacheExpiredTime: 500
+
+
+defaultLocale: ""
+
+localBundles: ""
+
+productName: JSCoreSDKTest
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/client/go/goclienttest/testdata/Config/configDefaultLocaleEmpty.yaml]: Has corrected it in new commit

File: autotest/client/go/goclienttest/testdata/localBundles/GoClientTest/1.0.0/contact/messages_de.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "component":"contact",
+    "messages":{
+      "contact.marketing": "Marketing:",
+      "contact.applicationname": "VIP-Sample-Web-Anwendung",
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/client/go/goclienttest/testdata/localBundles/GoClientTest/1.0.0/contact/messages_de.json]: Has corrected it in new commit

File: autotest/client/go/goclienttest/testdata/localBundles/GoClientTest2/1.0/c2/messages_en.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "component" : "c2",
+  "messages" : {
+    "demo.plural.users" : "{0, plural,one {Singleton Angular client has a user.} other {Singleton Angular client has # users.}}",
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/client/go/goclienttest/testdata/localBundles/GoClientTest2/1.0/c2/messages_en.json]: Has corrected it in new commit

File: autotest/client/go/goclienttest/LocalBundlesOnly/confignull.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+#unit: second
+cacheExpiredTime: 1
+
+
+defaultLocale: zh-Hans
+
+localBundles: "../testdata/localBundles/GoClientTest/"
+
+
+name: NoExistingProduct
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/client/go/goclienttest/LocalBundlesOnly/confignull.yaml]: Has corrected it in new commit

File: autotest/client/go/goclienttest/LocalBundlesOnly/getComponentTranslation_test.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+	cfPath := "config.yaml"
+	cfg, _ := sgtn.NewConfig(cfPath)
+	inst, _ := sgtn.NewInst(*cfg)
+	fmt.Print(inst)
+	translation := inst.GetTranslation()
+
+	Convey("component-localbundles-requestlocale: Get request locale's translation from localbundles", t, func() {
+
+		Convey("component-localbundle-requestlocale: request locale(fr) is in localebundle, return request locale translation from localbundles(P0)", func() {
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/client/go/goclienttest/LocalBundlesOnly/getComponentTranslation_test.go]: Has corrected it in new commit

File: autotest/client/go/goclienttest/LocalBundlesOnly/getStringTranslation_test.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+			tran2, _ := translation.GetStringMessage("fr", Defaultcom, htmlkey)
+			// should add check if send out http request as the previous case has cached fr component translation
+
+			So(tran2, ShouldEqual, frhtmlvalue)
+		})
+
+		Convey("localbundles-requestlocale: request a component(contact) is in localbundle but key isn't in localbundles, return key(P1)", func() {
+
+			tran2, _ := translation.GetStringMessage("fr", "contact", "non-existing.key")
```
Comment (Xiaochao8) [inline autotest/client/go/goclienttest/LocalBundlesOnly/getStringTranslation_test.go:56]:
```suggestion
			tran2, _ := translation.GetStringMessage("fr", "contact", "non-existent.key")
```
Change all the occurences of it?

**Pull request merged**